src/Discord/Internal/Types/Prelude.hs

```diff
@@ -197,8 +241,17 @@ class Data a => InternalDiscordEnum a where
         | fromIntegral (round i) == i = Just $ round i
         | otherwise = Nothing
 
-toMaybeJSON :: (ToJSON a) => a -> Maybe Value
-toMaybeJSON = return . toJSON
+-- when we upgrade aeson, we need to change Text to Key
+
```

Comment:
Which version is the problem? Could conditional compilation help>

Reply:
In my eyes, maintaining this for two separate major versions of something seems like too much of a hassle.

Reply:
A fair. We haven't gotten any requests to make the versions looser, so seems like people are happy with it.

It's a static hassle, once the conditional compilation is in place it works.

Reply:
I think the text vs key is the only breaking change Aeson2 makes. We could use the same conditional compilation  `# if MIN_VERSION_aeson(2, 0, 0)` as a template, for the `(.==)` code below. Wouldn't change anything else.

Try implementing the conditional compilation support with aeson 2. If it doesn't work well, I'll merge this PR as is (it's good). If the compilation works then I'd like it in though